datatable/java/datatable/src/test/java/io/cucumber/datatable/DataTableTypeRegistryTest.java

```diff
@@ -37,8 +46,132 @@ public Place transform(DataTable table) {
         }));
     }
 
+    @Test
+    public void returns_null_data_table_type_if_none_match_and_no_default_registered() {
+
+        registry.defineDataTableType(DataTableType.cell(Place.class));
+        registry.defineDataTableType(DataTableType.entry(Place.class));
+
+        DataTableType lookupTableTypeByType = registry.lookupTableTypeByType(constructType(Place.class));
+
+        assertNull(lookupTableTypeByType);
+    }
+
+    @Test
+    public void returns_null_data_table_type_for_cell_if_no_default_registered() throws Exception {
+
+        registerDefaultEntryTransformer();
+
+        DataTableType lookupTableTypeByType = registry.lookupTableTypeByType(constructTypeForCell());
+
+        assertNull(lookupTableTypeByType);
+    }
+
+    @Test
+    public void returns_null_data_table_type_for_entry_if_no_default_registered() throws Exception {
+
+        registerDefaultCellTransformer();
+
+        DataTableType lookupTableTypeByType = registry.lookupTableTypeByType(constructTypeForEntry());
+
+        assertNull(lookupTableTypeByType);
+    }
+
+
+
+    @SuppressWarnings({"unchecked", "ConstantConditions"})
+    @Test
+    public void returns_default_data_table_type_for_cell_if_none_match_and_default_registered() {
+
+        registerCellEntryAndDefaults();
+
+        String here="here";
+        DataTableType lookupTableTypeByTypeForCell = registry.lookupTableTypeByType(constructTypeForCell());
```

Comment:
From reading this test it is not immediately clear that `constructTypeForCell` creates a list of lists of Place type. Please use the same convention as used in `DataTableTypeRegistryTableConverterTest`.